### DISCUSSION TITLE
Updating an existing pv model from pvib 0.8.x to 0.9.1

### DISCUSSION THREAD

**Discussion by Ludee** [pvlib/pvlib-python#1475] [Q&A]
I am currently updating an existing pvlib model and having some problems with different errors.
In addition it is not clear what parts need to updated to run with the latest version.
The topics I found are:

- PVSystem module_type
- PVSystem racking_model
- TEMPERATURE_MODEL_PARAMETERS

Any help is appreciated. The result would be a step-by-step instruction to update your existing model.

**Comment by mikofski** [comment]
Can you please share a [minimum complete verifiable example](https://en.wikipedia.org/wiki/Minimal_reproducible_example)? Or a little more detail about each of the above specifically that you are having issues with?

**Reply by kandersolar** [reply]
@Ludee in case you aren't aware, we do log changes for each version in our "what's new?" documentation page: https://pvlib-python.readthedocs.io/en/stable/whatsnew.html

But if that doesn't answer all your questions we can help with specifics like @mikofski says.

**Reply by Ludee** [reply]
Thanks for the fast reply. Indeed my question is lacking a good example. sorry!
I am developing a model with different setups and weather data, thus I am not able to create a MWE that reproduces the errors.
The model is running with `pvlib 0.8.1` without any problems and with `pvlib 0.9.1`:

A:
> lib\site-packages\pvlib\tools.py:340: RuntimeWarning: divide by zero encountered in true_divide
  np.trunc(np.log(atol / (df['VH'] - df['VL'])) / np.log(phim1)))
> pvlib\tools.py:340: RuntimeWarning: invalid value encountered in log
  np.trunc(np.log(atol / (df['VH'] - df['VL'])) / np.log(phim1)))

B:
> lib\site-packages\pvlib\modelchain.py", line 1770, in run_model
    self._run_from_effective_irrad(weather)
> ...
> lib\site-packages\pvlib\tools.py", line 364, in _golden_sect_DataFrame
    raise Exception("Iterations exceeded maximum. Check that func",
Exception: ('Iterations exceeded maximum. Check that func', ' is not NaN in (lower, upper)')

Any further help is appreciated.

**Reply by cwhanse** [reply]
@Ludee between v0.8.1 and v0.9.1 there were some improvement to the code that computed the maximum power point on IV curves. The warnings and error you are encountering relate to that code. For the warnings (A), I would guess that at least one IV curve was requested with POA irradiance of 0, so photocurrent is 0, and Voc is zero. For the error, I'd guess something upstream produced a NaN for the IV curve Voc.

Without a reproducible example I couldn't diagnose further. I'm interested to know the cause of the error so we can improve the function if warranted.
